app/hall.hoon

```diff
@@ -1024,12 +1016,19 @@
       ^+  +>
       ::  only have presence if you have write permission.
       ?.  |((so-admire who) ?=($remove -.dif))  +>
-      ::  ignore if it won't result in change.
+      ::  ignore if it won't result in change,
+      ::  or if it sets an impersonating handle.
       ?.  ?:  ?=($remove -.dif)  (~(has by locals) who)
           ?|  !(~(has by locals) who)
             ::
-              =+  (~(got by locals) who)
-              !=(- (change-status - dif))
+              =+  old=(~(got by locals) who)
+              =+  new=(change-status - dif)
+              ?&  !=(old new)
+                ::
+                  ?=  $~
+                  (rush (fall han.man.new '') ;~(pfix sig fed:ag))
+                  ::TODO  %+ syntax errors?
```

Comment:
I don't understand this TODO. Does it mean that the call to `++rush` could error out instead of just producing nil?

Reply:
Attempting to call rush with a `%+` instead of the `()` shorthand there gives a syntax error for some reason, hence why this section looks a bit weird.

Reply:
Huh, that's weird. You should make a github issue for that. It's unrelated to this code, though.